src/cpu/pred/bpred_unit.cc

```diff
@@ -187,6 +192,13 @@ BPredUnit::predict(const StaticInstPtr &inst, const InstSeqNum &seqNum,
             tid, seqNum, hist->pc,  (hist->btbHit) ? "hit" : "miss");
 
 
+    // In a high performance CPU there is no other way than a BTB hit
+    // to know about a branch instruction. In that case consolidate
+    // indirect and RAS predictor only if there was a BTB it.
+    // For low end CPUs predecoding might be used to identify branches.
+    const bool allow_btb_override = (hist->btbHit || !requiresBTBHit);
```

Comment:
Optional: could this be named better? It doesn't really signal overriding the BTB (in fact, when true it uses the BTB result). Might something like `use_ras_and_indirect`, or `branch_type_info_available` be clearer to future readers?

Reply:
Agree on the confusing name. Why not using something like `branch_detected` as variable name?

Reply:
Thank you for the suggestions. I renamed the variable to `branch_detected`